### ISSUE TITLE
Cannot generate declaration file when using `ESLintUtils.RuleCreator`

### ISSUE THREAD

**Issue by roikoren755** [typescript-eslint/typescript-eslint#4198]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

Try to transpile the following file, with `"declaration": true` in the `tsconfig.json` file.

```TS
import { ESLintUtils } from "@typescript-eslint/experimental-utils";

export const createRule = ESLintUtils.RuleCreator((s) => s);
```

**Expected Result**

Transpilation should succeed, and the generated declaration file should work.

**Actual Result**

Transpilation fails, with the following error log:
```
index.ts:3:14 - error TS4023: Exported variable 'createRule' has or is using name 'RuleWithMetaAndName' from external module "/PATH/TO/PROJECT/node_modules/@typescript-eslint/experimental-utils/dist/eslint-utils/RuleCreator" but cannot be named.

3 export const createRule = ESLintUtils.RuleCreator((s) => s);
               ~~~~~~~~~~


Found 1 error.
```

**Additional Info**

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `5.4.0` |
| `TypeScript`                            | `4.5.2` |
| `node`                                  | `14.18.0` |

**Comment by JoshuaKGoldberg** [comment]
Ah, sorry about that @roikoren755! This is from my recent change #4136.
